src/Core/Services/ResolverMiddleware.cs

```diff
@@ -265,6 +265,10 @@ protected static bool IsInnerObject(IMiddlewareContext context)
                 SINGLE_TYPE => ((FloatValueNode)value).ToSingle(),
                 FLOAT_TYPE => ((FloatValueNode)value).ToDouble(),
                 DECIMAL_TYPE => ((FloatValueNode)value).ToDecimal(),
+                // If we reach here, we can be sure that the value will not be null.
+                // ID type can store int, string or Guid. First we try to parse the value as Guid.
+                // If the value is not a Guid, we just return the value.
+                ID_TYPE => Guid.TryParse(value.Value!.ToString(), out Guid guidValue) ? guidValue : value.Value,
```

Comment:
@Aniruddh25 cosmos can treat int values like 1,2,3.. etc for (fields like `id` shown in the image) and guids as well, as Hotchocolate ID:
<img width="697" alt="image" src="https://github.com/Azure/data-api-builder/assets/34566234/024110fb-750a-4837-8bf5-b8abe8c8108a">

whereas Sql only treats `System.Guid` type as Hotchocolate id.

So, when we find a field with Hotchocolate type as 'ID', we will first try to parse it as a GUID. If the parsing returned false, we will return the value.Value as it is.

This is kind of a fix for comos as well because for cosmos, if they try to have a GUID column today with hotchocolate type of 'ID', filters would not work, just like they don't work for Sql.

Reply:
I don't understand this assumption: "whereas Sql only treats System.Guid type as Hotchocolate id."

Your code comment captured an important point raised by GraphQL spec: 
> https://spec.graphql.org/October2021/#sel-FAHXZDCAOCEBpjG
> The ID scalar type represents a unique identifier, often used to refetch an object or as the key for a cache. The ID type is serialized in the same way as a String; however, it is not intended to be human-readable. While it is often numeric, it should always serialize as a String.

We follow this portion of the spec(assisted by HotChocolate serializing the field as a string), and then it is up to us (the server) to coerce the value into an "expected format". 

> [Result Coercion](https://spec.graphql.org/October2021/#sec-ID.Result-Coercion)
GraphQL is agnostic to ID format, and serializes to string to ensure consistency across many formats ID could represent, from small auto-increment numbers, to large 128-bit random numbers, to base64 encoded values, or string values of a format like [GUID](https://en.wikipedia.org/wiki/Globally_unique_identifier).
> 
> GraphQL services should coerce as appropriate given the ID formats they expect. When coercion is not possible they must raise a field error.
>
> [Input Coercion](https://spec.graphql.org/October2021/#sec-ID.Input-Coercion)
When expected as an input type, any string (such as "4") or integer (such as 4 or -4) input value should be coerced to ID as appropriate for the ID formats a given GraphQL service expects. Any other input value, including float input values (such as 4.0), must raise a request error indicating an incorrect type

What I don't understand with this PR's changes is that we are saying that a GUID column in MsSql **is a GraphQL ID field type**. However, such a column may not actually be a unique identifier for the particular row in the database, even though GUID is inherently a unique value.

Reply:
I see your point here. Apologies for the confusion but I meant that with the implementation I had, sql will treat guid as hotchocolate id type. You have a valid point here, a guid column can possibly not be the unique identifier aka ID of a record. Mapping of System.Guid to Hotchocolate.UUID makese sense, yes.